--- PKGBUILD.orig	2026-03-09 14:14:37.235782209 +0200
+++ PKGBUILD	2026-03-09 14:20:12.868993640 +0200
@@ -66,7 +66,34 @@ export KBUILD_BUILD_USER=$pkgbase
 export KBUILD_BUILD_TIMESTAMP="$(date -Ru${SOURCE_DATE_EPOCH:+d @$SOURCE_DATE_EPOCH})"
 
 prepare() {
-  cd $_srcname
+  # Out-of-tree module signing
+  #
+  ######################################################
+  # this is added at the start of prepare() & replaces #
+  # 'cd $_srcname'                                     #
+  ######################################################
+  # uncomment for linux-xanmod-cacule & some other AUR kernels
+  # to match the Package Maintainer's variable for the kernel
+  # sources directory.
+  #
+  # local _srcname=linux-${_major}
+
+  msg2 "Rebuilding local signing key..."
+
+  cp -rf /usr/src/certs-local ./
+  cd certs-local
+
+  msg2 "Updating kernel config with new key..."
+
+  # NB: config path must be quoted for file globbing to work
+  # some kernels have multiple config files (e.g linux-libre)
+  # to see configurable options run:
+  # /usr/src/certs-local/genkeys.py -h
+  ./genkeys.py -v --config '../config*'
+
+  cd ../$_srcname
+
+  # cd $_srcname
 
   echo "Setting version..."
   echo "-$pkgrel" > localversion.10-pkgrel
@@ -94,12 +121,12 @@ prepare() {
 build() {
   cd $_srcname
 
-  make htmldocs SPHINXOPTS=-QT &
-  local pid_docs=$!
+  # make htmldocs SPHINXOPTS=-QT &
+  # local pid_docs=$!
 
   make all
   make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
-  wait "${pid_docs}"
+  # wait "${pid_docs}"
 }
 
 _package() {
@@ -244,6 +271,33 @@ _package-headers() {
   echo "Adding symlink..."
   mkdir -p "$pkgdir/usr/src"
   ln -sr "$builddir" "$pkgdir/usr/src/$pkgbase"
+
+  # Out-of-tree module signing
+  ##################################################
+  # this is added at the end of _package-headers() #
+  ##################################################
+  # This is run in the kernel source / build directory
+  #
+  # some AUR kernels may also need to set $builddir to match
+  # the Package Maintainer's variable for the build dir
+  # inside the package:
+  #
+  # local builddir="$pkgdir/usr/lib/modules/${_kernver}/build"
+
+  msg2 "Local Signing certs for out-of-tree modules..."
+
+  certs_local_src="../certs-local"
+  certs_local_dst="${builddir}/certs-local"
+
+  # install certificates
+  ${certs_local_src}/install-certs.py $certs_local_dst
+
+  # install dkms tools
+  dkms_src="$certs_local_src/dkms"
+  dkms_dst="${pkgdir}/etc/dkms"
+  mkdir -p $dkms_dst
+
+  rsync -a $dkms_src/{kernel-sign.conf,kernel-sign.sh} $dkms_dst/
 }
 
 _package-docs() {
@@ -268,7 +322,7 @@ _package-docs() {
 pkgname=(
   "$pkgbase"
   "$pkgbase-headers"
-  "$pkgbase-docs"
+  # "$pkgbase-docs"
 )
 for _p in "${pkgname[@]}"; do
   eval "package_$_p() {

